pyvcloud/vcd/org.py

```diff
@@ -582,6 +608,39 @@ def get_roles_query(self, name_filter=None):
             qfilter=org_filter)
         return query, resource_type
 
+    def get_right(self, right_name):
+        """
+        Retrieves corresponding record of the specified right.
+        :param right_name: (str): The name of the right record to be retrieved
+        :return: (QueryResultRightRecordType): Right query result in records
+                 format
+        """  # NOQA
+        try:
+            rights_query = self.get_rights_query(('name', right_name))[0]
+            return rights_query.find_unique()
+        except MissingRecordException:
+            raise Exception('Right \'%s\' does not exist.' % right_name)
+
+    def get_rights_query(self, name_filter=None):
+        """
+        Get the typed query for the rights in the current Org
+        :param name_filter: (tuple): (name ,'right name') Filter the rights by
+                             'right name'
+        :return: (tuple of (_TypedQuery, str))
+                  _TypedQuery object represents the query for the rights in
+                  the current Org
+                  str represents the resource type of the query object
+        """  # NOQA
+        if self.resource is None:
+            self.resource = self.client.get_resource(self.href)
+
+        resource_type = 'right'
+        query = self.client.get_typed_query(
+            resource_type,
+            query_result_format=QueryResultFormat.RECORDS,
+            equality_filter=name_filter)
+        return query, resource_type
```

Comment:
Why are we passing resource_type here? I dont see it being used in the calling function.

Reply:
It is going to be used in another change-list underway (list rights in the org).

Reply:
I don't think we should be passing back resource_type unless we make get_rights_query a generic function. The name of the function is a dead giveaway that we are dealing with 'rights'.

Reply:
This should only return one value, the list of rights, see `list_catalogs` as an example